test/core/iomgr/ios/CFStreamTests/CFStreamEndpointTests.mm

```diff
@@ -89,17 +89,11 @@ - (BOOL)waitForEvent:(gpr_atm *)event timeout:(int)timeout {
   return (gpr_atm_acq_load(event) != -1);
 }
 
-+ (void)setUp {
-  grpc_init();
-}
-
-+ (void)tearDown {
-  grpc_shutdown();
-}
-
 - (void)setUp {
   self.continueAfterFailure = NO;
 
+  grpc_init();
```

Comment:
I wonder if grpc_init() is generally needed per test?

I'm slightly concerned that we might hide some code reentrancy issues when it's done per test. (but leaving to your judgement :))

Reply:
There have been issues in the past with performing `grpc_init`/`grpc_shutdown` and that is because grpc_shutdown can happen in an asynchronous manner causing weird issues when gRPC tries to get initialized again.

Reply:
got it, thanks Eryu/Yash . Removing the shutdown/init change here, seems like this might cause more flake if init/shutdown are invoked repeatedly per test.

Reply:
Thanks Denny. :)

In any case, for future changes if you were able to run the tests repeated for ~100 times and verify the flakiness before v.s. after it would be great to help make decisions :)

Reply:
yep exactly ;) I'm using the new test repetition feature on Xcode 13 beta (https://developer.apple.com/videos/play/wwdc2021/10296/) which can run these XCTest cases on dev machine x times

Reply:
Fantastic! (Can't believe it's such a new feature 😂)